.ci/releasenotes-gen-auto.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
```

Comment:
name it as `releasenotes-gen-xdoc-push.sh`

Reply:
Changes Done.

Reply:
I'll provide the result commit and local execution script soon.

Reply:
```

ASUS@RK7 MINGW64 ~/Desktop/opensource/checkstyle (auto-notes)
$ ./.ci/releasenotes-gen-xdoc-push.sh
Cloning into 'contribution'...
remote: Enumerating objects: 7655, done.
remote: Counting objects: 100% (929/929), done.
remote: Compressing objects: 100% (453/453), done.
remote: Total 7655 (delta 451), reused 779 (delta 382), pack-reused 6726
Receiving objects: 100% (7655/7655), 2.84 MiB | 3.16 MiB/s, done.
Resolving deltas: 100% (3909/3909), done.
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]
[INFO] -------------< com.github.checkstyle:releasenotes-builder >-------------
[INFO] Building releasenotes-builder 1.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ releasenotes-builder ---
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ releasenotes-builder ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ releasenotes-builder ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 19 source files to C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\target\classes
[INFO] /C:/Users/ASUS/Desktop/opensource/checkstyle/.ci-temp/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java: C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\src\main\java\com\github\checkstyle\github\NotesBuilder.java uses or overrides a deprecated API.
[INFO] /C:/Users/ASUS/Desktop/opensource/checkstyle/.ci-temp/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java: Recompile with -Xlint:deprecation for details.
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ releasenotes-builder ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ releasenotes-builder ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 19 source files to C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\target\classes
[INFO] /C:/Users/ASUS/Desktop/opensource/checkstyle/.ci-temp/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java: C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\src\main\java\com\github\checkstyle\github\NotesBuilder.java uses or overrides a deprecated API.
[INFO] /C:/Users/ASUS/Desktop/opensource/checkstyle/.ci-temp/contribution/releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java: Recompile with -Xlint:deprecation for details.
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ releasenotes-builder ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 16 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ releasenotes-builder ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\target\test-classes
[INFO] /C:/Users/ASUS/Desktop/opensource/checkstyle/.ci-temp/contribution/releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java: C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\src\test\java\com\github\checkstyle\templates\TemplateProcessorTest.java uses or overrides a deprecated API.
[INFO] /C:/Users/ASUS/Desktop/opensource/checkstyle/.ci-temp/contribution/releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java: Recompile with -Xlint:deprecation for details.
[INFO]
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ releasenotes-builder ---
[INFO] Surefire report directory: C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\target\surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.github.checkstyle.github.CommitMessageTest
Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.062 sec
Running com.github.checkstyle.templates.TemplateProcessorTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.455 sec

Results :

Tests run: 25, Failures: 0, Errors: 0, Skipped: 0

[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ releasenotes-builder ---
[INFO] Building jar: C:\Users\ASUS\Desktop\opensource\checkstyle\.ci-temp\contribution\releasenotes-builder\target\releasenotes-builder-1.0.jar
[INFO]
[INFO] --- maven-shade-plugin:2.4.2:shade (default) @ releasenotes-builder ---
[INFO] Including org.kohsuke:github-api:jar:1.99 in the shaded jar.
[INFO] Including org.apache.commons:commons-lang3:jar:3.9 in the shaded jar.
[INFO] Including commons-codec:commons-codec:jar:1.13 in the shaded jar.
[INFO] Including commons-io:commons-io:jar:2.6 in the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-databind:jar:2.12.6.1 in the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-annotations:jar:2.12.6 in the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-core:jar:2.12.6 in the shaded jar.
[INFO] Including org.eclipse.jgit:org.eclipse.jgit:jar:5.4.2.201908231537-r in the shaded jar.
[INFO] Including com.jcraft:jsch:jar:0.1.55 in the shaded jar.
[INFO] Including com.jcraft:jzlib:jar:1.1.1 in the shaded jar.
[INFO] Including com.googlecode.javaewah:JavaEWAH:jar:1.1.6 in the shaded jar.
[INFO] Including org.slf4j:slf4j-api:jar:1.7.2 in the shaded jar.
[INFO] Including org.bouncycastle:bcpg-jdk15on:jar:1.61 in the shaded jar.
[INFO] Including org.bouncycastle:bcprov-jdk15on:jar:1.61 in the shaded jar.
[INFO] Including org.bouncycastle:bcpkix-jdk15on:jar:1.61 in the shaded jar.
[INFO] Including org.slf4j:slf4j-simple:jar:1.7.12 in the shaded jar.
[INFO] Including com.google.guava:guava:jar:30.1.1-jre in the shaded jar.
[INFO] Including com.google.guava:failureaccess:jar:1.0.1 in the shaded jar.
[INFO] Including com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava in the shaded jar.
[INFO] Including com.google.code.findbugs:jsr305:jar:3.0.2 in the shaded jar.
[INFO] Including org.checkerframework:checker-qual:jar:3.8.0 in the shaded jar.
[INFO] Including com.google.errorprone:error_prone_annotations:jar:2.5.1 in the shaded jar.
[INFO] Including com.google.j2objc:j2objc-annotations:jar:1.3 in the shaded jar.
[INFO] Including org.freemarker:freemarker:jar:2.3.29 in the shaded jar.
[INFO] Including commons-cli:commons-cli:jar:1.4 in the shaded jar.
[INFO] Including commons-lang:commons-lang:jar:2.6 in the shaded jar.
[INFO] Including org.twitter4j:twitter4j-core:jar:4.0.7 in the shaded jar.
[INFO] Including javax.mail:mail:jar:1.5.0-b01 in the shaded jar.
[INFO] Including javax.activation:activation:jar:1.1 in the shaded jar.
[WARNING] jackson-annotations-2.12.6.jar, bcpkix-jdk15on-1.61.jar, bcpg-jdk15on-1.61.jar, bcprov-jdk15on-1.61.jar, jackson-core-2.12.6.jar, jackson-databind-2.12.6.1.jar define 1 overlapping classes:
[WARNING]   - module-info
[WARNING] maven-shade-plugin has detected that some class files are
[WARNING] present in two or more JARs. When this happens, only one
[WARNING] single version of the class is copied to the uber jar.
[WARNING] Usually this is not harmful and you can skip these warnings,
[WARNING] otherwise try to manually exclude artifacts based on
[WARNING] mvn dependency:tree -Ddetail=true and the above output.
[WARNING] See http://maven.apache.org/plugins/maven-shade-plugin/
[INFO] Attaching shaded artifact.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.123 s
[INFO] Finished at: 2022-06-20T06:42:17+05:30
[INFO] ------------------------------------------------------------------------
Cloning into 'checkstyle'...
remote: Enumerating objects: 208170, done.
remote: Counting objects: 100% (134/134), done.
remote: Compressing objects: 100% (104/104), done.
remote: Total 208170 (delta 38), reused 77 (delta 15), pack-reused 208036
Receiving objects: 100% (208170/208170), 81.46 MiB | 4.49 MiB/s, done.
Resolving deltas: 100% (119165/119165), done.
Updating files: 100% (4668/4668), done.
CS_RELEASE_VERSION=10.4-SNAPSHOT
LATEST_RELEASE_TAG=prepare-10.4.1

Execution succeeded!
releasenotes.xml after commit:
<?xml version="1.0" encoding="UTF-8"?>

<document xmlns="http://maven.apache.org/XDOC/2.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">

  <head>
    <title>Release Notes</title>
  </head>

  <body>
    <!-- placeholder for a new section -->


    <section name="Release 10.4-SNAPSHOT">
      <div class="releaseDate">20.06.2022</div>
      <p>Notes:</p>
        <ul>
          <li>
            Delete releasenotes-gen-xdocs-push.sh.
            Author: Rahul Khinchi
          </li>
          <li>
            Create releasenotes-gen-xdocs-push.sh.
            Author: Rahul Khinchi
          </li>
        </ul>
    </section>
    <section name="Release 10.3">
      <div class="releaseDate">29.05.2022</div>

ASUS@RK7 MINGW64 ~/Desktop/opensource/checkstyle (auto-notes)
$

```

Reply:
Please check out: https://github.com/checkstyle/checkstyle/pull/11750#issuecomment-1159863606

Reply:
I don't see how renaming related to execution

Reply:
Please do renaming of script

Reply:
@romani, this script is different as I used the same script but with different parameters like using Rahulkhinchi03/checkstyle and not checkstyle/checkstyle ( for checking if this is working or not).

I'll rename the file name. Please let me know if you need any change to be done. 

Did you see the command for the latest release tag?

Reply:
Please do not mix review items. I see all items that are not resolved

Reply:
File Renamed.